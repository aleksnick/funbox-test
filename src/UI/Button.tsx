import React from "react";
import UIButton from "@material-ui/core/Button";
import { Size } from "../Models/Size";

export interface SimpleButtonProps {
  color?: ButtonColor;
  href?: string;
  disabled?: boolean;
  onClick?: () => void;
}

export enum ButtonColor {
  Default = "default",
  Primary = "primary",
  Secondary = "secondary"
}

export enum ButtonVariant {
  Flat = "flat",
  Outlined = "outlined",
  Contained = "contained"
}

export interface ButtonProps extends SimpleButtonProps {
  variant?: ButtonVariant;
  size?: Size;
}

/**
 * Кнопка
 *
 * @export
 * @class Button
 * @extends {React.Component<ButtonProps>}
 */
export default class Button extends React.Component<ButtonProps> {
  constructor(props: ButtonProps) {
    super(props);
  }

  render() {
    const { variant, color, size, href, disabled, children } = this.props;
    return (
      <UIButton
        variant={variant || ButtonVariant.Outlined}
        color={color || ButtonColor.Default}
        disabled={!!disabled}
        href={href}
        size={size || Size.Small}
        onClick={this.onClick}
      >
        {children}
      </UIButton>
    );
  }

  onClick = (e: React.ChangeEvent<{}>) => {
    const onClick = this.props.onClick;
    if (onClick) {
      e.preventDefault();
      e.stopPropagation();
      onClick();
    }
  };
}
