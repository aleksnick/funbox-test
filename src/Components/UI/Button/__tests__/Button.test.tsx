import React from "react";
import Button from "../";
import { render } from "enzyme";

describe("Button", () => {
  describe("Snapshots", () => {
    test("Simple Button snapshot", () => {
      const wrapper = render(<Button>Button</Button>);
      expect(wrapper).toMatchSnapshot();
    });
  });
});
