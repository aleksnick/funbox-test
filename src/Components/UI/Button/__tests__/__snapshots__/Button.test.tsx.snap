// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button Snapshots Simple Button snapshot 1`] = `
<button
  class="MuiButtonBase-root-27 MuiButton-root-1 MuiButton-outlined-9 MuiButton-sizeSmall-24"
  tabindex="0"
  type="button"
>
  <span
    class="MuiButton-label-2"
  >
    Button
  </span>
</button>
`;
